include/seqan/bam_io/bam_alignment_record.h

```diff
@@ -607,6 +607,37 @@ getAlignmentLengthInRef(BamAlignmentRecord const & record)
     return l;
 }
 
+// ----------------------------------------------------------------------------
+// Function appendRawPod()
+// ----------------------------------------------------------------------------
+
+#if SEQAN_BIG_ENDIAN
+inline void
+ensure_little_endian(BamAlignmentRecordCore & r)
+{
+    ensure_little_endian(r.rID);
+    ensure_little_endian(r.beginPos);
+    // _l_qname unchanged because 8bit
+    // mapQ unchanged because 8bit
+    r.bin       = htole16(r.bin);
```

Comment:
That is kind of weird to call the specific system function here.
I know that these are bit fields pointing to `uint32_t` types, but maybe you could add a internal function that fixes the byte size and let `makeLittleEndian` in the default implementation call the internals.
```cpp
namespace detail
{
template <typename TInt>
auto makeLittleEndian(TInt const value, ByteSize<4> const &)
{
    return htole32(value);
}
template <typename TInt>
auto makeLittleEndian(TInt const value, ByteSize<2> const &)
{
    return htole16(value);
}
...

template <typename T>
auto makeLittleEndian(T const value)
{
    return makeLittleEndian(value, ByteSize<sizeof(T)>{});
}
}
```

You can also keep passing it by reference.

Reply:
I don't understand what you mean, maybe we can discuss this?

Reply:
This won't work, because you cannot take the address of bitfields. There is no way around manually reassigning this here. One could use `enforceLittleEndian` instead of `htole16` like this:
```cpp
auto tmp1 = r.bin;
enforceLittleEndian(tmp1);
r.bin = tmp1;
```
But it would very likely prevent optimisations here. `htole16` is actually standardised so it may not be pretty, but I think it is ok to use it in this one scenario.